tests/test_upload.py

```diff
@@ -87,7 +92,53 @@ def get_password(service_name, username):
 def test_get_repository_keyring():
     with modified_env({'FLIT_PASSWORD': None}), \
             _fake_keyring('tops3cret'):
-        repo = upload.get_repository(cfg_file=io.StringIO(pypirc2))
+        repo = upload.get_repository(pypirc_path=io.StringIO(pypirc2))
 
     assert repo['username'] == 'fred'
     assert repo['password'] == 'tops3cret'
+
+
+pypirc3_repo = "https://invalid-repo.inv"
+pypirc3_user = "test"
+pypirc3_pass = "not_a_real_password"
+pypirc3 = f"""
+[distutils] =
+index-servers =
+    test123
+
+[test123]
+repository: {pypirc3_repo}
+username: {pypirc3_user}
+password: {pypirc3_pass}
+"""
+
+
+def test_upload_pypirc_file(copy_sample):
+    with patch('os.path.isfile', return_value=True), \
```

Comment:
I don't particularly like having to mock this out. I think we should decide whether the functions accept file-like objects, and make that work without having to mock anything, or take only file paths, and test using temp files.

I'd say use temp files. Pytest makes that pretty convenient, and the code under test shouldn't really have special additional code paths just for the tests.

Reply:
Ok, will change the approach to using temp files.

Reply:
Done, please let me know what you think. Thanks!